vpr/src/base/vpr_api.cpp

```diff
@@ -369,6 +373,25 @@ void vpr_init_with_options(const t_options* options, t_vpr_setup* vpr_setup, t_a
     device_ctx.pad_loc_type = vpr_setup->PlacerOpts.pad_loc_type;
 }
 
+/** Port equivalence does not make sense during flat routing.
+ * Remove port equivalence from all ports in the architecture */
+static void unset_port_equivalences(DeviceContext& device_ctx){
+    for(auto& physical_type: device_ctx.physical_tile_types){
```

Comment:
Add an output message saying you're doing this.
Have a somewhat more expansive comment (why are you doing this?) at the declaration.